message_tagging_service/config.py

```diff
@@ -28,23 +28,145 @@
+        :rtype: str
+        """
+        config_file = os.environ.get('MTS_CONFIG_FILE')
+        if config_file:
+            return config_file
+        elif 'MTS_DEV' in os.environ or running_tests:
+            # Use {project root directory}/conf/config.py
+            return os.path.realpath(
+                os.path.join(os.path.dirname(__file__), '..', 'conf', 'config.py'))
+        else:
+            if profile:
+                return f'/etc/mts/config.{profile}.py'
+            else:
+                return '/etc/mts/config.py'
+
+    @staticmethod
+    def get_config_class_name():
+        """
+        Determine a class name according to running mode
+        :return: class name
+        :rtype: str
+        """
+        if running_tests:
+            return 'TestConfiguration'
+        elif 'MTS_DEV' in os.environ:
+            return 'DevConfiguration'
+        else:
+            return 'BaseConfiguration'
+
+    def load_new_config(self, profile=None, config_file=None, config_class=None,
+                        keep_overrides=False):
+        """
+        Reloads a base file from a config class
+        :param profile: to be set to config file name
+        :type profile: str
+        :param config_file: force configuration file path (overrides profile setting if present)
+        :type config_file: str
+        :param config_class: force class name in configuration file
+        :type config_class: str
+        :param keep_overrides: keep overrided values if True
+        :type keep_overrides: bool
+        :return:
+        """
+        if not config_file:
+            config_file = self.get_config_file(profile=profile)
+        loader = importlib.machinery.SourceFileLoader('mts_conf', config_file)
+        mod = loader.load_module()
+        if not config_class:
+            config_class = self.get_config_class_name()
+        if getattr(mod, config_class, None) is not None:
+            self.__conf_class = getattr(mod, config_class)
+        else:
+            raise AttributeError(f'Configuration class {config_class} '
+                                 f'not found in configuration file {config_file}')
+        if not keep_overrides:
+            self.__overrides.clear()
+
+    def __getattr__(self, item):
+        """
+        Allows to access configuration parameters as instance.item
+        :param item: item name
+        :type item: str
+        :return: item value
+        """
+        # check overrides first:
+        if self.__overrides.get(item) is not None:
+            return self.__overrides[item]
+        # trying configuration class then
+        if getattr(self.__conf_class, item) is not None:
+            return getattr(self.__conf_class, item)
+        # fallback to defaults if any
+        if self.__defaults.get(item) is not None:
+            return self.__defaults[item]
+        raise KeyError('Parameter {par_name} not found in configuration.'.format(par_name=item))
+
+    def __getitem__(self, item):
```

Comment:
This looks like a copy of ``__getattr__``. Can we merge them to avoid duplicate code?

Reply:
Done